core/src/main/java/com/orientechnologies/orient/core/metadata/schema/OType.java

```diff
@@ -108,13 +94,15 @@
 
   LINKBAG("LinkBag", 22, ORidBag.class, new Class<?>[] { ORidBag.class }),
 
-  ANY("Any", 23, null, new Class<?>[] {});
+  ANY("Any", 23, null, new Class<?>[] {}),
+
+  DELTA_RECORD("DeltaRecord", 24, ODocumentDelta.class, new Class<?>[] { ODocumentDelta.class });

```

Comment:
The OType are the types allowed as value in OrientDB, define a new OType implies that you can use it as a field type or as schema property type, and as soon as you define a new OType this need to be supported everywhere (from binary protocol to query engine), I understand the use case for this new type in the diff serialization, but probably is less impacting define specific serialization types just inside the diff specification, like for example: FULL_CONTENT, DIFF_CONTENT, REMOVED  and leave the OTypes unchanged.

Reply:
I agree with this. 
Just one explanation. Related to deltas there are two types of type. One is operation type  (core/src/main/java/com/orientechnologies/orient/core/record/impl/UpdateDeltaValueType.java). This one is equivalent to your example FULL_CONTENT, DIFF_CONTENT, ...
The second one is value type and it is equivalent to OType, but it has one more data type to handle, and that is Delta Record type.
So as OType has very specific purpose I've introduced new interface (core/src/main/java/com/orientechnologies/orient/core/metadata/schema/OTypeInterface.java) and new enum (core/src/main/java/com/orientechnologies/orient/core/delta/ODeltaDocumentFieldType.java) just for deltas